### PR TITLE
make non-interactive `support-bundle` output more machine readable

### PR DESCRIPTION
**Pull request by danbudris**
when using the `interactive=false` flag of `support-bundle`, the spinner would still spin and the archive path and analysis output were kind of smooshed together with the logs. This made it harder to parse the output programatically. RE: https://github.com/replicatedhq/troubleshoot/issues/418

Now, if `--interactive=false`, `support-bundle` only prints each received collector callback message once, and the spinner does not spin 🎡 

Also,  this adds a key to the archivePath and analyzerOutput that are returned, `archivePath` and `analyzerOutput`, for easier programatic parsing. Note that `archivePath` is also appended to the output for `--interactive=true`.

### Sample Output
The analyzer results are abbreviated for clarity, but I think it tells the whole story. This basically just prints each message the collector sends to the callback channel, so for each collection operation it performs, from my reading (hence the 13 printings of "Collecting support bundle logs").

```bash
dan@dansMachine:~/git/supportbundle$ support-bundle bundle.yaml --kubeconfig ./cluster.kubeconfig --interactive=false
Collecting support bundle cluster-info
Collecting support bundle cluster-resources
W0916 00:56:37.916499  442923 warnings.go:70] extensions/v1beta1 Ingress is deprecated in v1.14+, unavailable in v1.22+; use networking.k8s.io/v1 Ingress
Collecting support bundle secret
Collecting support bundle logs
Collecting support bundle logs
Collecting support bundle logs
Collecting support bundle logs
Collecting support bundle logs
Collecting support bundle logs
Collecting support bundle logs
Collecting support bundle logs
Collecting support bundle logs
Collecting support bundle logs
Collecting support bundle logs
Collecting support bundle logs
Collecting support bundle logs

analyzerResults=[
    {
        "name": "capv.controller.manager.status",
        "labels": {
            "desiredPosition": "1",
            "iconKey": "kubernetes_deployment_status",
            "iconUri": "https://troubleshoot.sh/images/analyzer-icons/deployment-status.svg?w=17\u0026h=17"
        },
        "insight": {
            "name": "capv.controller.manager.status",
            "labels": {
                "iconKey": "kubernetes_deployment_status",
                "iconUri": "https://troubleshoot.sh/images/analyzer-icons/deployment-status.svg?w=17\u0026h=17"
            },
            "primary": "capv-controller-manager Status",
            "detail": "capv-controller-manager is running.",
            "severity": "debug"
        },
        "severity": "debug",
        "analyzerSpec": ""
    },
    {
        "name": "capv.controller.manager.status",
        "labels": {
            "desiredPosition": "1",
            "iconKey": "kubernetes_deployment_status",
            "iconUri": "https://troubleshoot.sh/images/analyzer-icons/deployment-status.svg?w=17\u0026h=17"
        },
        "insight": {
            "name": "capv.controller.manager.status",
            "labels": {
                "iconKey": "kubernetes_deployment_status",
                "iconUri": "https://troubleshoot.sh/images/analyzer-icons/deployment-status.svg?w=17\u0026h=17"
            },
            "primary": "capv-controller-manager Status",
            "detail": "capv-controller-manager is running.",
            "severity": "debug"
        },
        "severity": "debug",
        "analyzerSpec": ""
    },
]
archivePath=support-bundle-2021-09-16T00_56_27.tar.gz
```

### Testing
`support-bundle bundle.yaml --kubeconfig ./mycluster.kubeconfig`. The spinner spins as expected, output is much the same, interactive terminal output still works + looks great.

`support-bundle bundle.yalm --kubeconfig ./mycluster.kubeconfig --interactive=false`. This spinner is gone, each callback msg is printed once, and the output is more easily machine parsable.

### CODE REVIEW

**Comment by DanStough** [comment]
First off, thank you so much @danbudris for this contribution and being so thorough in your description!

Since we're trying to fix the output here to be machine-parsable, what if:

1. We disable all progress notifications when `interactive=false`.
2. Instead of the top-level `analyzerResults=` and `archivePath=`:
```json
{
    "analyzerResults": [
        {
            "name": "capv.controller.manager.status",
            "labels": {
                "desiredPosition": "1",
                "iconKey": "kubernetes_deployment_status",
                "iconUri": "https://troubleshoot.sh/images/analyzer-icons/deployment-status.svg?w=17\u0026h=17"
            },
            "insight": {
                "name": "capv.controller.manager.status",
                "labels": {
                    "iconKey": "kubernetes_deployment_status",
                    "iconUri": "https://troubleshoot.sh/images/analyzer-icons/deployment-status.svg?w=17\u0026h=17"
                },
                "primary": "capv-controller-manager Status",
                "detail": "capv-controller-manager is running.",
                "severity": "debug"
            },
            "severity": "debug",
            "analyzerSpec": ""
        },
        {
            "name": "capv.controller.manager.status",
            "labels": {
                "desiredPosition": "1",
                "iconKey": "kubernetes_deployment_status",
                "iconUri": "https://troubleshoot.sh/images/analyzer-icons/deployment-status.svg?w=17\u0026h=17"
            },
            "insight": {
                "name": "capv.controller.manager.status",
                "labels": {
                    "iconKey": "kubernetes_deployment_status",
                    "iconUri": "https://troubleshoot.sh/images/analyzer-icons/deployment-status.svg?w=17\u0026h=17"
                },
                "primary": "capv-controller-manager Status",
                "detail": "capv-controller-manager is running.",
                "severity": "debug"
            },
            "severity": "debug",
            "analyzerSpec": ""
        }
    ],
    "archivePath": "support-bundle-2021-09-16T00_56_27.tar.gz"
}
```

This would get us to the point where you could run `kubectl support-bundle spec.yaml --interactive=false | jq <some filter here>`. There's some precedence [here](https://github.com/replicatedhq/troubleshoot/blob/fe4db40b43a35c60ca1472e1803f61c7b7867c77/cmd/preflight/cli/stdout_results.go#L40-L82) in preflight checks, though I don't think we are necessarily looking for parity with preflights (and especially not in just this PR). What do you think?

**Comment by danbudris** [comment]
Hey Dan, thanks for the review. And yeah, that seems right -- I don't see a ton of value in the progress notifications, and having it all in JSON is the move from a machine readability standpoint. Will update hopefully tomorrow

**Comment by danbudris** [comment]
allllright, that looks a lot better....tho as you can see there are deprecation warnings from k8s being printed, I assume somewhere in the collection process as it interacts with the api server. Will need to dig a bit deeper and see where in the process those are surfacing. 

```bash
W0917 14:59:00.489664  502211 warnings.go:70] extensions/v1beta1 Ingress is deprecated in v1.14+, unavailable in v1.22+; use networking.k8s.io/v1 Ingress
W0917 14:59:00.563633  502211 warnings.go:70] extensions/v1beta1 Ingress is deprecated in v1.14+, unavailable in v1.22+; use networking.k8s.io/v1 Ingress
W0917 14:59:00.638282  502211 warnings.go:70] storage.k8s.io/v1beta1 StorageClass is deprecated in v1.19+, unavailable in v1.22+; use storage.k8s.io/v1 StorageClass
W0917 14:59:01.019396  502211 warnings.go:70] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
{
    "analysis": [
        {
            "name": "capv.controller.manager.status",
            "labels": {
                "desiredPosition": "1",
                "iconKey": "kubernetes_deployment_status",
                "iconUri": "https://troubleshoot.sh/images/analyzer-icons/deployment-status.svg?w=17\u0026h=17"
            },
            "insight": {
                "name": "capv.controller.manager.status",
                "labels": {
                    "iconKey": "kubernetes_deployment_status",
                    "iconUri": "https://troubleshoot.sh/images/analyzer-icons/deployment-status.svg?w=17\u0026h=17"
                },
                "primary": "capv-controller-manager Status",
                "detail": "capv-controller-manager is running.",
                "severity": "debug"
            },
            "severity": "debug",
            "analyzerSpec": ""
        },
        {
            "name": "capv.controller.manager.status",
            "labels": {
                "desiredPosition": "1",
                "iconKey": "kubernetes_deployment_status",
                "iconUri": "https://troubleshoot.sh/images/analyzer-icons/deployment-status.svg?w=17\u0026h=17"
            },
            "insight": {
                "name": "capv.controller.manager.status",
                "labels": {
                    "iconKey": "kubernetes_deployment_status",
                    "iconUri": "https://troubleshoot.sh/images/analyzer-icons/deployment-status.svg?w=17\u0026h=17"
                },
                "primary": "capv-controller-manager Status",
                "detail": "capv-controller-manager is running.",
                "severity": "debug"
            },
            "severity": "debug",
            "analyzerSpec": ""
        },
    ],
    "archivePath": "support-bundle-2021-09-17T14_58_48.tar.gz"
}
```

File: cmd/troubleshoot/cli/run.go [hunk 1]
```diff
@@ -265,3 +269,28 @@ func canTryInsecure() bool {
...
 }
+
+type analysisOutput struct {
+	Analysis    []*analyzer.AnalyzeResult
+	ArchivePath string
+}
+
+func (a *analysisOutput) FormattedAnalysisOutput() (outputJson string, err error) {
+	type convertedOutput struct {
+		ConvertedAnalysis []*convert.Result `json:"analysis"`
```
Comment (danbudris) [inline cmd/troubleshoot/cli/run.go]:
```suggestion
		ConvertedAnalysis []*convert.Result `json:"analyzerResults"`
```

**Review by divolgin — APPROVED** [review]

**Review by DanStough — APPROVED** [review]
@danbudris thank you for the updates. I have some code that will suppress the `client-go` warnings and remove the control characters. We can merge as-is.

**Pull request merged**